Cannot install qiskit-qec on a Windows machine: `ERROR: Failed building editable for qiskit_qec`

### Steps to reproduce the problem
Attempt to install the project in editable mode following the [installation guide](https://github.com/kaiott/qiskit-qec/blob/main/docs/installation.md). The line `pip install -e .` will fail.
**Operating system**: Windows 11
**CMake version**: 3.28.0-msvc1
**C++ Compiler**: Visual C++ 2022 (version 17.9.0)

### What is the current behavior?
Installing dependencies: ok
Installation fails at `Building wheels for collected packages: qiskit_qec`: `error: subprocess-exited-with-error`
`ERROR: Failed building editable for qiskit_qec`
`Failed to build qiskit_qec`
`ERROR: Could not build wheels for qiskit_qec, which is required to install pyproject.toml-based projects`

The error stack is rather long and contains various additional details and warnings. The first error encountered is:
```
C:\Users\<username>\qiskit-qec\src\qiskit_qec\analysis\intern\distance.cpp(147,30): error C2039: 'inserter': is not 
a member of 'std' [C:\Users\<username>\AppData\Local\Temp\tmpbcn03w6u.build-temp\Release\qiskit_qec.circuits._c_circuits\src
\qiskit_qec\analysis\libanalysis.vcxproj]
        C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\include\stdexcept(19,1):
        see declaration of 'std'
```

### What is the expected behavior?
The installation process should complete without fatal errors on Windows 11, when following the installation guide.

--- a/src/qiskit_qec/analysis/distance.h
+++ b/src/qiskit_qec/analysis/distance.h
@@ -5,6 +5,7 @@
 #include <vector>
 #include <set>
 #include <exception>
+#include <iterator>
 
 #include "linear.h"
 #include "combinations.h"

--- a/src/qiskit_qec/analysis/distance.py
+++ b/src/qiskit_qec/analysis/distance.py
@@ -113,14 +113,12 @@ def _minimum_distance_2_python(stabilizer: np.ndarray, gauge: np.ndarray, max_we
     for row in range(xl.shape[0]):
         for w in range(1, max_weight + 1):
             if _distance_test(stabilizer.astype(int), xl[row].astype(int), w):
-                if w < weight:
-                    weight = w
+                weight = min(weight, w)
                 break
     for row in range(zl.shape[0]):
         for w in range(1, max_weight + 1):
             if _distance_test(stabilizer.astype(int), zl[row].astype(int), w):
-                if w < weight:
-                    weight = w
+                weight = min(weight, w)
                 break
     if weight < max_weight + 1:
         return weight

--- a/src/qiskit_qec/decoders/decoding_graph.py
+++ b/src/qiskit_qec/decoders/decoding_graph.py
@@ -322,7 +322,7 @@ class DecodingGraph:
 
         boundary_nodes = []
         for n, node in enumerate(self.graph.nodes()):
-            if node.is_boundary:
+            if node.is_logical:
                 boundary_nodes.append(n)
 
         for edge in self.graph.edge_list():

